include/pinocchio/multibody/model-graph.hpp

```diff
@@ -0,0 +1,766 @@
+//
+// Copyright (c) 2025 INRIA
+//
+
+#ifndef __pinocchio_multibody_model_graph_hpp__
+#define __pinocchio_multibody_model_graph_hpp__
+
+#include "pinocchio/multibody/fwd.hpp"
+#include "pinocchio/multibody/joint/joint-generic.hpp"
+#include "pinocchio/multibody/model.hpp"
+#include "pinocchio/spatial/inertia.hpp"
+#include "pinocchio/spatial/se3.hpp"
+
+#include <Eigen/Core>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/depth_first_search.hpp>
+#include <boost/graph/visitors.hpp>
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <unordered_map>
+
+namespace pinocchio
+{
+  struct JointFixedGraph
+  {
+    JointFixedGraph() = default;
+  };
+
+  struct JointRevoluteGraph
+  {
+    // rotation axis
+    Eigen::Vector3d axis;
+
+    JointRevoluteGraph(const Eigen::Vector3d & ax)
+    : axis(ax)
+    {
+    }
+  };
+
+  struct JointRevoluteUnboundedGraph
+  {
+    Eigen::Vector3d axis;
+
+    JointRevoluteUnboundedGraph(const Eigen::Vector3d & ax)
+    : axis(ax)
+    {
+    }
+  };
+
+  struct JointPrismaticGraph
+  {
+    Eigen::Vector3d axis;
+    JointPrismaticGraph(const Eigen::Vector3d & ax)
+    : axis(ax)
+    {
+    }
+  };
+
+  struct JointFreeFlyerGraph
+  {
+    JointFreeFlyerGraph() = default;
+  };
+
+  struct JointSphericalGraph
+  {
+    JointSphericalGraph() = default;
+  };
+
+  // Flipped whne model is reversed ?
+  struct JointSphericalZYXGraph
+  {
+    JointSphericalZYXGraph() = default;
+  };
+
+  struct JointTranslationGraph
+  {
+    JointTranslationGraph() = default;
+  };
+
+  struct JointPlanarGraph
+  {
+    JointPlanarGraph() = default;
+  };
+
+  struct JointHelicalGraph
+  {
+    Eigen::Vector3d axis;
+    double pitch;
+
+    JointHelicalGraph(const Eigen::Vector3d & ax, const double & p)
```

Comment:
```suggestion
    JointHelicalGraph(const Eigen::Vector3d & ax, double p)
```
Pas besoin de ref pour un type primitif.